frontend/src/app/user-interface/burger-menu/burger-menu.component.ts

```diff
@@ -0,0 +1,38 @@
+import { Component, OnInit } from '@angular/core';
+import {MatMenuModule, MatButtonModule, MatIconModule} from '@angular/material';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+
+@Component({
+  selector: 'app-burger-menu',
+  templateUrl: './burger-menu.component.html',
+  styleUrls: ['./burger-menu.component.scss']
+})
+export class BurgerMenuComponent implements OnInit {
+  lastSelected:string = "";
+  constructor() { }
+
+  ngOnInit() {
+  }
+    onClick(event: Event){
+        let elementClass;
+        if(event.srcElement.children.length != 0)
+            elementClass = event.srcElement.id.toString().replace(/button_/g,'');
+        else
+            elementClass = ( <HTMLElement>event.target ).parentElement.id.toString().replace(/button_/g,'');
```

Comment:
po zmianie html to ten if-else będzie niepotrzebny bo nigdy nie klikniemy w parenta + nie będziemy musieli wyciągać z id tego co szukamy

Reply:
zrobione